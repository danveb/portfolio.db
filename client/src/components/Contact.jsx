import "../styles/Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="contact__wrapper">
                <div className="contact__main">
                    <div className="contact__title">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact__intro">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                        <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact