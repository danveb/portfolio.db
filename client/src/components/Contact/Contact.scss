.contact {
    height: calc(100vh - 50px); 
    display: flex; 
    flex-direction: column; 
    background-color: var(--alice-blue); 

    .container {
        height: 70vh; 
        margin: 30px 30px 0 30px; 
        background-color: var(--alice-blue); 

        .contact-wrapper {
            height: 100%; 
            padding: 40px 60px; 
            display: flex; 
            justify-content: center; 
            align-items: center; 
            flex-direction: column; 

            @media screen and (max-width: 480px) {
                padding: 20px; 
            }

            .contact-headline {

                h1 {
                    color: var(--raisin-black); 
                    font-size: 40px; 
                    margin: 10px 0;

                    span {
                        color: var(--bottle-green); 
                    }

                    @media screen and (max-width: 768px) {
                        font-size: 30px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 20px; 
                    }
                }
            }

            .contact-introduction {
                max-width: 768px; 
                width: 50%; 

                @media screen and (max-width: 768px) {
                    width: 75%; 
                }
                
                @media screen and (max-width: 480px) {
                    width: 100%; 
                }

                p {
                    color: var(--cadet-blue-crayola); 
                    text-align: center; 
                    margin: 10px 0;
                    font-size: 18px; 

                    @media screen and (max-width: 768px) {
                        font-size: 16px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 14px; 
                    }
                }
            }

            .cta {
                display: flex; 
                justify-content: center; 
                align-items: center; 
                flex-direction: row; 
                margin: 20px 0; 
                border: 1px solid var(--bottle-green); 
                border-radius: 55px; 
                padding: 15px 25px; 
                background-color: var(--raisin-black); 
                color: var(--alice-blue); 
                text-transform: uppercase; 
                cursor: pointer; 
                text-decoration: none; 
                
                @media screen and (max-width: 768px) {
                    margin: 15px 0; 
                    padding: 10px 20px; 
                    font-size: 14px; 
                }
                
                @media screen and (max-width: 480px) {
                    font-size: 12px; 
                    margin: 10px 0;
                }
            }
        }
    }
}