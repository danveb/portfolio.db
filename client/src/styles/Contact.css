.contact {
  width: 100%; 
  height: 100%; 
}

.contact__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
}

.contact__main {
  height: 100%; 
  width: 100%; 
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 40px 0; 
  position: relative; 
}

.contact__title {
  font-family: "Mulish", sans-serif;
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.contact__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.sidebar {
  background-color: rgba(35, 55, 75, 0.9);
  color: #fff;
  padding: 6px 10px;
  font-family: monospace;
  z-index: 2;
  position: absolute;
  margin: 12px;
  font-size: 10px; 
}

.map-container {
  height: 45vh; 
  width: 100%; 
  margin-bottom: 40px; 
}

@media screen and (max-width: 1024px) {
  .contact__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .contact__title {
    font-size: 16px; 
  }

  .contact__intro {
    font-size: 14px; 
  }
}