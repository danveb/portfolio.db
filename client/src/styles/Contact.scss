.contact {
    background-color: var(--alice-blue); 

    .contact__wrapper {
        max-width: 1200px; 
        height: 100%; 
        margin: 30px auto; 

        .contact__main {
            display: flex; 
            flex-direction: column; 
            justify-content: center; 
            align-items: center; 
            max-width: 590px; 
            margin: auto; 

            .contact__title {
                h1 {
                    color: var(--raisin-black); 
                    font-size: 38px; 
                    margin: 10px 0; 

                    span {
                        color: var(--bottle-green); 
                    }
                }
            }

            .contact__intro {
                width: 75%; 
                color: var(--raisin-black); 
                display: flex; 
                flex-direction: column; 
                justify-content: center; 
                align-items: center; 

                p {
                    color: var(--cadet-blue-crayola); 
                    text-align: center; 
                    margin: 20px 0;
                    font-size: 18px; 

                    @media screen and (max-width: 768px) {
                        font-size: 16px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 14px; 
                    }
                }

                .cta {
                    margin: 20px 0; 
                    border: 1px solid var(--bottle-green); 
                    border-radius: 55px; 
                    padding: 15px 25px; 
                    background-color: var(--raisin-black); 
                    color: var(--alice-blue); 
                    text-transform: uppercase; 
                    cursor: pointer; 
                    text-decoration: none; 
                    
                    @media screen and (max-width: 768px) {
                        margin: 15px 0; 
                        padding: 10px 20px; 
                        font-size: 14px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 12px; 
                        margin: 10px 0;
                    }
                }
            }
        }
    }
}