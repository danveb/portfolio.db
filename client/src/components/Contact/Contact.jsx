import { socials } from "../../constants/socials"; 
import "./Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="container">
                <div className="contact-wrapper">
                    <div className="contact-headline">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact-introduction">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                    </div>
                    <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                </div>
            </div>
            <div className="footer">
                <div className="footer-wrapper">
                    <div className="footer-headline">
                        <h1>Social Links<span>.</span></h1>
                    </div>
                    <div className="footer-social">
                        <ul>
                            {socials.map((social) => (
                                <li key={social.id}>
                                    <a href={social.url} target={social.target} rel={social.rel}>{<social.icon />}</a>
                                </li>
                            ))}
                        </ul>
                    </div>
                    <div className="footer-mr">
                        <p>Portfolio design & development - Danny Bae</p>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact