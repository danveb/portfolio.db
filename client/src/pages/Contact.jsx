import React, { useRef, useState, useEffect } from "react"; 
import mapboxgl from "!mapbox-gl" // eslint-disable-line import/no-webpack-loader-syntax
import "../styles/Contact.css"; 
import { Button } from "../components";

mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_API; 

export default function Contact() {
  // initialize mapbox state/ref
  const mapContainer = useRef(null); 
  const map = useRef(null); 
  const [lat, setLat] = useState(41.8781);
  const [lng, setLng] = useState(-87.6298); 
  const [zoom, setZoom] = useState(11); 

  // useEffect
  useEffect(() => {
    if(map.current) return; // initialize map just once
    map.current = new mapboxgl.Map({
      container: mapContainer.current,
      style: "mapbox://styles/mapbox/dark-v11", 
      center: [lng, lat], 
      zoom: zoom, 
    }); 

    if(!map.current) return; // wait for map to initialize 
    map.current.on("move", () => {
      setLng(map.current.getCenter().lng.toFixed(4));
      setLat(map.current.getCenter().lat.toFixed(4)); 
      setZoom(map.current.getZoom().toFixed(2)); 
    });
  }, [lat, lng, zoom]); 

  return (
    <div className="contact">
      <div className="contact__wrapper">
        <div className="contact__main">
          <div className="contact__title">
            <h1>send me a note<span className="dot--green">.</span> üìù</h1>
          </div>
          <div className="breakline"></div>
          <div className="contact__intro">
            <p>I'm in the Chicago area, but open to discussing new opportunities to work together regardless of location. Feel free to send a note and I'll do my best to get back to you. </p>
          </div>
          <Button title="let's email" url="mailto:danibae91@gmail.com?subject=üë®‚Äçüíª  hey Danny, let's work together!" />
        </div>
        <div className="sidebar">
        Lng: {lng} | Lat: {lat}
        </div>
        <div ref={mapContainer} className="map-container" />
      </div>
    </div>
  )
}