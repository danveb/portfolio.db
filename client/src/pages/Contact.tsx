import { MapboxGL, Note } from "../components";
import "../styles/Contact.css"; 

export default function Contact() {
  return (
    <div className="contact">
      <div className="contact__wrapper">
        <section className="contact__main">
          <div className="contact__title">
            <h1>send me a note<span className="dot--green">.</span> üìù</h1>
          </div>
          <div className="breakline"></div>
          <div className="contact__intro">
            <p>I'm in the Chicago area, but open to discussing new opportunities to work together regardless of location. Feel free to send a note and I'll do my best to get back to you. </p>

          </div>
        </section>
        <MapboxGL />
        <Note />
      </div>
    </div>
  )
}