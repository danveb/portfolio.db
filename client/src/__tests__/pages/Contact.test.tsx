import { render, screen } from "@testing-library/react"; 
import Contact from "../../pages/Contact"; 
import { BrowserRouter } from "react-router-dom";

describe("Contact component", () => {
  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Contact />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });

    // getByText
  test("should display the title", () => {
    render(
      <BrowserRouter>
        <Contact />
      </BrowserRouter>
    ); 
    const title = screen.getByText("send me a note", {
      exact: false
    }); 
    expect(title).toBeInTheDocument(); 
  });

  // getByText 
  test("should display a p tag", () => {
    render(
      <BrowserRouter>
        <Contact />
      </BrowserRouter>
    ); 
    const pTag = screen.getByText("Chicago", { exact: false }); 
    expect(pTag).toBeInTheDocument(); 
  });
});