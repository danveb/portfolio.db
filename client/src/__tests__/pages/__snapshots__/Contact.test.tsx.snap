// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="contact"
  >
    <div
      class="contact__wrapper"
    >
      <section
        class="contact__main"
      >
        <div
          class="contact__title"
        >
          <h1>
            send me a note
            <span
              class="dot--green"
            >
              .
            </span>
             üìù
          </h1>
        </div>
        <div
          class="breakline"
        />
        <div
          class="contact__intro"
        >
          <p>
            I'm in the Chicago area, but open to discussing new opportunities to work together regardless of location. Feel free to send a note and I'll do my best to get back to you. 
          </p>
        </div>
      </section>
      <div
        class="note"
      >
        <div
          class="note__wrapper"
        >
          <form>
            <label
              for="user_name"
            >
              Name
            </label>
            <input
              id="user_name"
              name="user_name"
              placeholder="Your name"
              required=""
              type="text"
            />
            <label
              for="user_email"
            >
              Email
            </label>
            <input
              id="user_email"
              name="user_email"
              placeholder="Your email"
              required=""
              type="email"
            />
            <label
              for="message"
            >
              Message
            </label>
            <textarea
              id="message"
              name="message"
              placeholder="Any message you'd like to send"
              required=""
            />
            <input
              class="submitBtn"
              type="submit"
              value="shoot email üë®‚Äçüíª"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
